dialogue-core/src/main/java/com/palantir/dialogue/core/AimdConcurrencyLimiter.java

```diff
@@ -120,25 +122,40 @@ void dropped() {
          */
         void success() {
             inFlight.decrementAndGet();
-            int newLimit = limit.accumulateAndGet(inFlightSnapshot, LimitUpdater.SUCCESS);
+            double newLimit = accumulateAndGet(limit, inFlightSnapshot, LimitUpdater.SUCCESS);
             log.info("UP {}", SafeArg.of("newLimit", newLimit));
         }
     }
 
-    enum LimitUpdater implements IntBinaryOperator {
+    private static double accumulateAndGet(
+            AtomicDouble atomicDouble, double argument, DoubleBinaryOperator accumulatorFunction) {
+        double prev = atomicDouble.get();
+        double next = 0;
+        for (boolean haveNext = false; ; ) {
+            if (!haveNext) {
+                next = accumulatorFunction.applyAsDouble(prev, argument);
+            }
+            if (atomicDouble.compareAndSet(prev, next)) {
+                return next;
+            }
+            haveNext = (prev == (prev = atomicDouble.get()));
+        }
+    }
+
+    enum LimitUpdater implements DoubleBinaryOperator {
         SUCCESS() {
             @Override
-            public int applyAsInt(int originalLimit, int inFlightSnapshot) {
-                if (inFlightSnapshot >= originalLimit * 0.95) {
-                    return Math.min(MAX_LIMIT, originalLimit + 1);
+            public double applyAsDouble(double originalLimit, double inFlightSnapshot) {
+                if (inFlightSnapshot >= originalLimit * BACKOFF_RATIO) {
+                    return Math.min(MAX_LIMIT, originalLimit + (1D / originalLimit));
```

Comment:
Wanna put a little comment to say we intentionally want to do (proportionally) bigger increments when our concurrency limit is low (i.e. jumping from 1 -> 2 in one hop is desirable), but we want to be a bit more cautious about jumping from 100 -> 101?